library/std/src/sys/pal/uefi/time.rs

```diff
@@ -112,6 +123,52 @@ pub(crate) mod system_time_internal {
 
         Duration::new(utc_epoch, t.nanosecond)
     }
+
+    /// This algorithm is taken from: http://howardhinnant.github.io/date_algorithms.html
+    pub(crate) const fn uefi_time_from_duration(
+        dur: crate::time::Duration,
+        daylight: u8,
+        timezone: i16,
+    ) -> r_efi::system::Time {
+        const SECS_IN_MINUTE: u64 = 60;
+        const SECS_IN_HOUR: u64 = SECS_IN_MINUTE * 60;
+        const SECS_IN_DAY: u64 = SECS_IN_HOUR * 24;
+
+        let secs = if timezone == r_efi::efi::UNSPECIFIED_TIMEZONE {
+            dur.as_secs()
+        } else {
+            dur.as_secs().checked_add_signed(-timezone as i64).unwrap()
```

Comment:
I'd use subtraction here, just like the formula in the UEFI spec.
```suggestion
            dur.as_secs().checked_sub_signed(timezone as i64).expect("times should be representable as local UEFI times")
```

Reply:
`checked_sub_signed` is currently unstable. I can add the feature if that is fine though.